<b>I was having difficulty getting agate working so I decided to give Pandas a try</b>

In [387]:
import csv
import pandas as pd
import numpy as np

In [388]:
low_memory =False

Uploading the file

In [389]:
hatecrime = pd.read_csv('oncampushate111213.csv')

<b>Given that some of the recent incidents on MU's campus have involved vandalism, that's what
I've decided to focus on. First, I thought I would learn how many schools were in each state.</b>

In [390]:
campus_per_state = hatecrime[['State']]
print campus_per_state.groupby(['State']).count().sort(['State'], ascending=False)

       State
State       
CA      1059
TX       729
FL       695
NY       628
PA       492
OH       455
IL       444
GA       349
MI       331
MO       325
NC       320
VA       284
TN       272
AZ       248
MA       240
IN       230
PR       210
NJ       207
KY       203
WA       199
WI       196
OK       190
LA       188
MN       183
CO       174
SC       173
IA       149
OR       143
MD       141
KS       133
AR       124
AL       122
CT       116
UT       105
MS        95
WV        88
NE        84
NM        82
NV        68
ID        64
ME        61
NH        58
DC        46
HI        45
ND        42
MT        39
DE        36
SD        35
RI        32
VT        32
AK        32
WY        19
FM         6
GU         3
VI         3
MP         3
MH         2
AS         1
PW         1

[59 rows x 1 columns]


<b>There are 325 schools from Missouri in the database. Good to know.<b> 

<b>Now I'll try to make a general table for ten Missouri schools, starting with the largest body of students.</b>

In [391]:
mo_schools = hatecrime[['UNITID_P', 'INSTNM', 'BRANCH', 'City', 'State', 'men_total', 'women_total', 'Total']]
mo_schools_population = mo_schools.loc[mo_schools['State'] == 'MO'].groupby(['INSTNM']).aggregate(np.sum).sort(['Total'], ascending=False)
print mo_schools_population.dropna()[:10]

                                                                         UNITID_P  \
INSTNM                                                                              
Saint Louis Community College                                          1613772052   
Webster University                                                     1619046326   
Columbia College                                                       1593585139   
Metropolitan Community College-Kansas City                             1067970021   
Lindenwood University                                                  1423744048   
Central Methodist University-College of Graduate and Extended Studies  7569539153   
Park University                                                        1251047148   
Ozarks Technical Community College                                      887360015   
Kaplan University-Davenport Campus                                      260901004   
Southeast Missouri State University                              

<b>Hmmm...so, MU Columbia is not within the top ten in terms of the student body population. 
Then again, the record layout doesn't really specify what makes up the grand total.
Oh well, let's try to see if we can get the data on Mizzou by searching for city and state.</b>

In [392]:
schools = hatecrime[['UNITID_P', 'INSTNM', 'BRANCH', 'City', 'State', 'men_total', 'women_total', 'Total']]
mo_schools = schools.loc[schools['State'] == 'MO']
mo_schools_columbia = mo_schools.loc[mo_schools['City'] == 'Columbia'].groupby(['INSTNM']).aggregate(np.sum).sort(['Total'], ascending=False)
print mo_schools_columbia.dropna()[:10]

                                                                        UNITID_P  \
INSTNM                                                                             
Columbia College                                                       177065001   
Moberly Area Community College                                         178448002   
Central Methodist University-College of Graduate and Extended Studies  445267004   
William Woods University                                               179964002   
Bryan University                                                       475583001   

                                                                       men_total  \
INSTNM                                                                             
Columbia College                                                            6867   
Moberly Area Community College                                              2249   
Central Methodist University-College of Graduate and Extended Studies      

<b>No Mizzou or Stephens College. Let's try something else.</b>

<b>Lets look for the 20 most populous schools in the entire database.</b>

In [393]:
big_schools = schools.groupby(['INSTNM']).aggregate(np.sum).sort(['Total'], ascending=False)
print big_schools.dropna()[:20]

                                                        UNITID_P  men_total  \
INSTNM                                                                        
Ivy Tech Community College                            3623688300     981216   
Embry-Riddle Aeronautical University-Worldwide       57978715157    1711152   
Webster University                                   11873006895     500082   
Houston Community College                             4283037259     466906   
Austin Community College District                     5351808373     444240   
University of Central Florida                         2126448151     429504   
Northern Virginia Community College                   3261244105     353640   
New York University                                   3102400145     297904   
Northern Arizona University                           2422590316     248377   
Columbia College                                      7439736713     241826   
Miami Dade College                                  

<b>Now, let's look at the first 20 schools that had the largest number of reported vandalism cases from 2011 - 2012</b>

In [394]:
hatecrime['VANDAL_111213'] = hatecrime['VANDAL11'] + hatecrime['VANDAL12'] + hatecrime['VANDAL13']
vandalism = hatecrime[['INSTNM', 'State', 'BRANCH', 'VANDAL11', 'VANDAL12', 'VANDAL13', 'VANDAL_111213']]
van_inst = vandalism.sort(['VANDAL_111213'], ascending=False)
print van_inst.dropna()[:20]

                                         INSTNM State  \
1023                         Fort Lewis College    CO   
5742                    Delaware Valley College    PA   
1146                            Trinity College    CT   
469           University of California-Berkeley    CA   
5906  Pennsylvania State University-Main Campus    PA   
3160                    Northeastern University    MA   
1183                        American University    DC   
2220               Indiana University-Southeast    IN   
5350                            Oberlin College    OH   
5365          Ohio State University-Main Campus    OH   
2961        University of Maryland-College Park    MD   
4721                             SUNY at Albany    NY   
1087                  University of Connecticut    CT   
482         University of California-Santa Cruz    CA   
4723                         SUNY at Binghamton    NY   
2713                        Thomas More College    KY   
1133                      Quinn

<b>Taking a quick look at the top 3, Fort Lewis in Colorado had the highest number of vandlism incidents during this time period with 15, with incidents increasing with each passing year. Delaware Valley in Pennsylvania was second with 14, with incidents decreasing with each year and Trinity College in Conneticut is number 3 with 12 reported incidents and a pretty massive decrease over the same period of time. Now to get the national total for reported vandalism, year by year over this period.</b> 

In [395]:
van11total = hatecrime['VANDAL11'].sum()
print 'Below is the number of cases of vandalism reported on college campuses in 2011'
print van11total
van12total = hatecrime['VANDAL12'].sum()
print 'Below is the number of cases of vandalism reported on college campuses in 2012'
print van12total
van13total = hatecrime['VANDAL13'].sum()
print 'Below is the number of cases of vandalism reported on college campuses in 2013'
print van13total
van3year_total = van11total + van12total + van13total
print 'The total from 2011, 2012, and 2013 is:'
print van3year_total

Below is the number of cases of vandalism reported on college campuses in 2011
371.0
Below is the number of cases of vandalism reported on college campuses in 2012
410.0
Below is the number of cases of vandalism reported on college campuses in 2013
370
The total from 2011, 2012, and 2013 is:
1151.0


<b>Reported incidents of vandalism spiked up in 2012 with 410 reported incidents from the previous year's 371 before dropping back down to 370 in 2013. Over this period there were 1151 reported incidents of vandalism.</b> 

In [396]:
hatecrime['VANDAL_RAC111213'] = hatecrime['VANDAL_RAC11'] + hatecrime['VANDAL_RAC12'] + hatecrime['VANDAL_RAC13']
vandal_race111213 = hatecrime['VANDAL_RAC111213'].sum()
print 'Vandalism by racial bias over three years:'
print vandal_race111213

hatecrime['VANDAL_REL111213'] = hatecrime['VANDAL_REL11'] + hatecrime['VANDAL_REL12'] + hatecrime['VANDAL_REL13']
vandal_religion111213 = hatecrime['VANDAL_REL111213'].sum()
print 'Vandalism by religious bias over three years:'
print vandal_religion111213

hatecrime['VANDAL_SEX111213'] = hatecrime['VANDAL_SEX11'] + hatecrime['VANDAL_SEX12'] + hatecrime['VANDAL_SEX13']
vandal_sex111213 = hatecrime['VANDAL_SEX111213'].sum()
print 'Vandalism by sexual orientation bias over three years:'
print vandal_sex111213

hatecrime['VANDAL_GEN111213'] = hatecrime['VANDAL_GEN11'] + hatecrime['VANDAL_GEN12'] + hatecrime['VANDAL_GEN13']
vandal_gender111213 = hatecrime['VANDAL_GEN111213'].sum()
print 'Vandalism by gender bias over three years:'
print vandal_gender111213

hatecrime['VANDAL_DIS111213'] = hatecrime['VANDAL_DIS11'] + hatecrime['VANDAL_DIS12'] + hatecrime['VANDAL_DIS13']
vandal_disability111213 = hatecrime['VANDAL_DIS111213'].sum()
print 'Vandalism by disability bias over three years:'
print vandal_disability111213

hatecrime['VANDAL_ETH111213'] = hatecrime['VANDAL_ETH11'] + hatecrime['VANDAL_ETH12'] + hatecrime['VANDAL_ETH13']
vandal_ethnicity111213 = hatecrime['VANDAL_ETH111213'].sum()
print 'Vandalism by ethnicity bias over three years:'
print vandal_ethnicity111213

vandal_hate_all = vandal_race111213 + vandal_religion111213 + vandal_sex111213 + vandal_gender111213 + vandal_disability111213 + vandal_ethnicity111213
print 'All together:'
print vandal_hate_all

Vandalism by racial bias over three years:
502.0
Vandalism by religious bias over three years:
175.0
Vandalism by sexual orientation bias over three years:
325.0
Vandalism by gender bias over three years:
34.0
Vandalism by disability bias over three years:
3.0
Vandalism by ethnicity bias over three years:
100.0
All together:
1139.0


<b>So there were a total of 1139 incidents of vandalism on college campuses over this three year period classified as a hatecrime. Given that the record layout for this is a little unclear, this either means that out of a total of 1151 incidents of vandalism, 98% of these reported incidents were classified as a hate crime OR that there were almost as many hate-crime related incidents of vandalism as ordinary acts of vandalism on college campuses.

44 percent of all hate-based vandalism on campuses over this three year period was tied to a racial bias. Nearly 29 percent was over sexual orientation, 15 percent involved religious bias, and nearly 9 percent involved a relgious bias.
</b>

<b>Breaking down the reported hate crimes for each year</b>

In [397]:
print 'Reported incidents per year, 2011, 2012, 2013'
print 'In 2011 to 2013, racial vandalism incidents:' 
print hatecrime['VANDAL_RAC11'].sum()
print hatecrime['VANDAL_RAC12'].sum()
print hatecrime['VANDAL_RAC13'].sum()

print 'Religion, 2011 - 2013:'
print hatecrime['VANDAL_REL11'].sum()
print hatecrime['VANDAL_REL12'].sum()
print hatecrime['VANDAL_REL13'].sum()

print 'Sexual Orientation 2011 -2013:' 
print hatecrime['VANDAL_SEX11'].sum()
print hatecrime['VANDAL_SEX12'].sum()
print hatecrime['VANDAL_SEX13'].sum()

print 'Gender 2011-2013'
print hatecrime['VANDAL_GEN11'].sum()
print hatecrime['VANDAL_GEN12'].sum()
print hatecrime['VANDAL_GEN13'].sum()

print 'Disability 2011-2013'
print hatecrime['VANDAL_DIS11'].sum()
print hatecrime['VANDAL_DIS12'].sum()
print hatecrime['VANDAL_DIS13'].sum()

print 'Ethnicity 2011-2013'
print hatecrime['VANDAL_ETH11'].sum()
print hatecrime['VANDAL_ETH12'].sum()
print hatecrime['VANDAL_ETH13'].sum()


Reported incidents per year, 2011, 2012, 2013
In 2011 to 2013, racial vandalism incidents:
171.0
187.0
152
Religion, 2011 - 2013:
57.0
71.0
48
Sexual Orientation 2011 -2013:
105.0
107.0
114
Gender 2011-2013
7.0
11.0
17
Disability 2011-2013
1.0
0.0
2
Ethnicity 2011-2013
30.0
34.0
37


<b>So, there's not a lot of change in the number of incidents over the years. Vandalism tied to racial bias on college campuses remained the highest for each year, although it did drop by around 19 percent in 2013 and religious-related vandalism dropped by about 32 percent the same year. Vandalism tied to sexual orientation rose over this period.</b>

In [398]:
hatecrime['VANDAL_HATE111213'] = hatecrime['VANDAL_RAC111213'] + hatecrime['VANDAL_REL111213'] + hatecrime['VANDAL_SEX111213'] + hatecrime['VANDAL_GEN111213'] + hatecrime['VANDAL_DIS111213'] + hatecrime['VANDAL_ETH111213']
hate_by_state = hatecrime[['State', 'VANDAL_RAC111213', 'VANDAL_REL111213', 'VANDAL_SEX111213', 'VANDAL_GEN111213', 'VANDAL_DIS111213', 'VANDAL_ETH111213', 'VANDAL_HATE111213']]
hate_by_state_result = hate_by_state.groupby(['State']).aggregate(np.sum).sort(['VANDAL_HATE111213'], ascending=False)
print hate_by_state_result

       VANDAL_RAC111213  VANDAL_REL111213  VANDAL_SEX111213  VANDAL_GEN111213  \
State                                                                           
NY                   67                47                23                 7   
CA                   50                11                32                 3   
PA                   36                20                27                 1   
MA                   18                12                21                 1   
OH                   24                 4                13                 2   
CT                   16                 8                20                 0   
IL                   21                 3                12                 1   
MD                   22                 8                 9                 0   
NJ                   17                12                10                 0   
MI                   15                 8                 7                 0   
WA                   12     

<b>I thought this may have yielded something interesting, but it's a little underwhelming. Both NY and CA have a lot of campuses.</b>

In [399]:
hate_by_campus = hatecrime[['INSTNM', 'State', 'BRANCH', 'VANDAL_RAC111213', 'VANDAL_REL111213', 'VANDAL_SEX111213', 'VANDAL_GEN111213', 'VANDAL_DIS111213', 'VANDAL_ETH111213', 'VANDAL_HATE111213']]
hate_by_campus_result = hate_by_campus.sort(['VANDAL_HATE111213'], ascending=False)
print hate_by_campus_result.dropna()[:10]

                                         INSTNM State  \
1023                         Fort Lewis College    CO   
5742                    Delaware Valley College    PA   
1146                            Trinity College    CT   
469           University of California-Berkeley    CA   
5906  Pennsylvania State University-Main Campus    PA   
3160                    Northeastern University    MA   
1183                        American University    DC   
2220               Indiana University-Southeast    IN   
5350                            Oberlin College    OH   
5365          Ohio State University-Main Campus    OH   

                           BRANCH  VANDAL_RAC111213  VANDAL_REL111213  \
1023                  Main Campus                 2                 1   
5742      Delaware Valley College                 6                 4   
1146  Main Campus Trinity College                 0                 3   
469                   Main Campus                 1                 1   
5906   

<b>While this is based around hate-based vandalism incidents, it's important to note that this table is almost identical to the earlier table based on reported vandalism. This gives credence to the notion that the VANDAL_11 column and the subsequent years bundle the total of hate-based vandalism and other vandalism together.</b>

In [400]:
hate_by_campus_ny = hate_by_campus.loc[hate_by_campus['State'] == 'NY'].sort(['VANDAL_HATE111213'], ascending=False)
print hate_by_campus_ny.dropna()[:10]

                                           INSTNM State  \
4723                           SUNY at Binghamton    NY   
4721                               SUNY at Albany    NY   
4850           SUNY Westchester Community College    NY   
4730                           Buffalo State SUNY    NY   
4506                             Hamilton College    NY   
4725                       Stony Brook University    NY   
4426  Columbia University in the City of New York    NY   
4381                                 Bard College    NY   
4414                            Cazenovia College    NY   
4734    State University of New York at New Paltz    NY   

                                                 BRANCH  VANDAL_RAC111213  \
4723                                        Main Campus                 2   
4721                                        Main Campus                 6   
4850                                        Main Campus                 7   
4730                              Buffalo 

<b>Could potentially make for an interesting infographic. What kind of incidents have occurred on each campus over this three-year period. Ex. SUNY Albany has had more racially motivated incidents of vandalism while SUNY Binghamton has had more instances of relgious vandalism.</b>

In [401]:
hate_by_campus_mo = hate_by_campus.loc[hate_by_campus['State'] == 'MO'].sort(['VANDAL_HATE111213'], ascending=False)
print hate_by_campus_mo.dropna()[:10]

                                          INSTNM State  \
3809             University of Missouri-Columbia    MO   
3880                      Saint Louis University    MO   
3892               Saint Louis Community College    MO   
3916              North Central Missouri College    MO   
3875                    Ranken Technical College    MO   
3989                      William Jewell College    MO   
3899                Southwest Baptist University    MO   
3769                       Lindenwood University    MO   
3782  Metropolitan Community College-Kansas City    MO   
3825                       Neosho Beauty College    MO   

                                     BRANCH  VANDAL_RAC111213  \
3809                            Main Campus                 4   
3880                            Main Campus                 0   
3892  Saint Louis Community College-Meramec                 2   
3916                            Main Campus                 0   
3875                            Main

<b>Ah, there's Mizzou at number one for the number of hate-based vandalism on a college campus in Missouri. There were four reported incidents at Mizzou during this period, all racially based. There were only sixteen reported cases of hate-based vandalism on Missouri campuses during this three-year period.</b>

In [402]:
MU_VANDAL_RACE = hatecrime[['INSTNM', 'VANDAL_RAC11', 'VANDAL_RAC12', 'VANDAL_RAC13']]
mizzou_hate = MU_VANDAL_RACE.loc[MU_VANDAL_RACE['INSTNM'] == 'University of Missouri-Columbia']
print mizzou_hate.dropna()

                               INSTNM  VANDAL_RAC11  VANDAL_RAC12  \
3809  University of Missouri-Columbia             2             0   

      VANDAL_RAC13  
3809             2  

[1 rows x 4 columns]


<b>And now I know when the incidents occurred which will make it just a little bit easier if I were to request the incident reports of these events.</b>

<b>While I think the idea has potential, I'm not quite sure if much of what's been pulled from this database would work for a local piece. The incidents that have occurred at Mizzou could make for a decent contextual piece to current events, but I'm not sure this database is much use for that. 

Racially motivated incidents of vandalism occur on campuses more often than other hate-based vandalism incidents. Based on what I've uncovered, it looks like incidents of hate-based vandalism vastly overshadow non hate-based incidents. While there appears to have been some drops in religious and racial incidents, gender, sexual orientation, and ethnicity incidents increased over the three-year period.

I could see the information yielded from this database featured as supplemental material to a more personal story about the economic cost (which I would guess is usually marginal) and the emotional stress these incidents create on a college campus. Readers could visit a page and look up the number of incidents on the campus they enter in the search box. This data could also potentially be used for mapping incidents of vandalism and color-coding them by their particular classification (race, religion, sexual orientation, gender, etc.). </b>